### ISSUE TITLE
Option "useProtoFieldName" to generate interfaces in snake_case

### ISSUE THREAD

**Issue by doochik** [timostamm/protobuf-ts#137]
I use protobuf-ts in my project like this
Example proto
```proto
syntax = "proto3";

message MyMessage {
  int32 foo_bar = 1;
}
```

Output interface after compilation 
```ts
export interface MyMessage {
    fooBar: number;
    /* I want to have
      foo_bar: number;
   */
}
```

Usage
```ts
const message = MyMessage.fromBinary(buffer, { readUnknownField: false });
const result =  MyMessage.toJson(message, {
    emitDefaultValues: false,
    useProtoFieldName: true,
});
```

`result` looks like  `{ foo_bar: 1 }`. But I can't cast it `const result: MyMessage` because `MyMessage` has fields in camelCase.

Are you agree to add such option to `ts_opt`? If yes I can help you and create PR for that :)
Thanks!

**Comment by timostamm** [comment]
Would this have any technical benefit or is it a code style preference?

**Comment by doochik** [comment]
It’s just a codestyle

**Comment by timostamm** [comment]
snake_case is the recommend style for protobuf field names. For generated code, the consensus seems to be to adopt the languages style. For ES, that seems to be camelCase. 

I think it's best to stick with that. I won't rule out the option, but am very wary of adding that feature now, because it will definitely be ballast for future development.

**Comment by doochik** [comment]
Sad to hear it. I understand your point. But I can't rewrite my very huge codebase at once.

At this time generated JSON and generated types are different and unusable. So I have to fork and fix it on my side

**Comment by timostamm** [comment]
> At this time generated JSON and generated types are different

That will probably always be true, though. JSON can't map many things properly / in a way that is convenient to use. A bit more detail in https://github.com/timostamm/protobuf-ts/issues/111#issuecomment-855290682 

[protoc-gen-tsjson](https://github.com/LLKennedy/protoc-gen-tsjson) generates just TypeScript interfaces for the JSON representation. Maybe something that could help? It doesn't support `emitDefaultValus` yet unfortunately.

> So I have to fork and fix it on my side

👍

**Comment by doochik** [comment]
Thanks, I’ll take a look to use ts-proto to generate interfaces

**Comment by jcready** [comment]
This has been addressed in https://github.com/timostamm/protobuf-ts/pull/250 which was released in [v2.3.0](https://github.com/timostamm/protobuf-ts/releases/tag/v2.3.0).
